HelpMyStreetFE/HelpMyStreetFE/Controllers/RegistrationController.cs

```diff
@@ -185,84 +185,19 @@ public async Task<ActionResult> StepThreePost([FromForm] StepThreeFormModel form
 
                 await _groupMemberService.AddUserToDefaultGroups(user.ID);
 
-                return Redirect("/registration/step-four");
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError("Error executing step 3");
-                _logger.LogError(ex.ToString());
-                return Redirect("/registration/step-three?failure=error");
-            }
-        }
-
-        [HttpGet("[controller]/step-four")]
-        public async Task<ActionResult> StepFour(CancellationToken cancellationToken)
-        {
-            var user = await _authService.GetCurrentUser(HttpContext, cancellationToken);
-
-            string correctPage = await GetCorrectPage(user);
-            if (!correctPage.StartsWith("/registration/step-four"))
-            {
-                // A different step needs to be completed at this point
-                return Redirect(correctPage);
-            }
-
-            var nearby = await _addressService.GetPostcodeDetailsNearUser(user);
-
-            var userPostcode = nearby.Where(p => p.Postcode == user.PostalCode).FirstOrDefault();
-            var nearbyToDisplay = nearby.Where(p => p.Postcode != user.PostalCode).OrderBy(p => p.ChampionCount).ThenBy(p => p.DistanceInMetres).Take(4).ToList();
-
-            // Insert user postcode at the top of the list...
-            nearbyToDisplay.Insert(0, userPostcode);
-
-            // ...but re-sort if there are already 2 champions there
-            if (userPostcode.ChampionCount >= 2)
-            {
-                nearbyToDisplay = nearbyToDisplay.OrderBy(p => p.ChampionCount).ThenBy(p => p.DistanceInMetres).ToList();
-            }
-
-            var championsNeeded = nearbyToDisplay.Aggregate(false, (acc, next) => acc || next.ChampionCount < 2);
-
-            return View(new StepFourRegistrationViewModel
-            {
-                ActiveStep = 4,
-                NearbyPostCodes = nearbyToDisplay,
-                UsersPostCode = userPostcode,
-                LocalAvailability = championsNeeded
-            });
-        }
-
-        [HttpPost("[controller]/step-four")]
-        public async Task<ActionResult> StepFourPost([FromForm] StepFourFormModel form, CancellationToken cancellationToken)
-        {
-            var user = await _authService.GetCurrentUser(HttpContext, cancellationToken);
-            string correctPage = await GetCorrectPage(user);
-            if (!correctPage.StartsWith("/registration/step-four"))
-            {
-                // A different step needs to be completed at this point
-                return Redirect(correctPage);
-            }
-
-            try
-            {
-                if (!form.ChampionRoleUnderstood)
-                {
-                    form.ChampionPostcodes = new System.Collections.Generic.List<string>();
-                }
-                _logger.LogInformation($"Step 4 submission for {user.ID}");
                 await _userService.CreateUserStepFourAsync(
```

Comment:
Ideally we should remove the call to _userService.CreateUserStepFourAsync.  Does the User Service complain if we do that?

Reply:
To make things not complain, I think a tweak will be needed to GetRegistrationIsComplete in UserService.cs